### ISSUE TITLE
Monorepo only builds root project when builder set to swc

### ISSUE THREAD

**Issue by clintonb** [nestjs/nest#12977]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

I followed the instructions at https://docs.nestjs.com/cli/monorepo to create a basic monorepo. I added an app named `worker` and set its port to `4444` (instead of `3000`) to easily identify its compiled output. The output is correct when `nest-cli.json` has builder set to either `webpack` (default) or `tsc`.

Only the _root_ app is built when this value is set to `swc`, regardless of what name I pass to `nest build <name>`.

### Minimum reproduction code

https://github.com/clintonb/nest-monorepo-swc

### Expected behavior

I expect the app(s) to be compiled with SWC when `builder` is set to `swc`.

If this is not supported, clearly document this fact so others don't waste time wondering if their projects are somehow misconfigured.

### Package

- [X] I don't know. Or some 3rd-party package
- [ ] <code>@nestjs/common</code>
- [ ] <code>@nestjs/core</code>
- [ ] <code>@nestjs/microservices</code>
- [ ] <code>@nestjs/platform-express</code>
- [ ] <code>@nestjs/platform-fastify</code>
- [ ] <code>@nestjs/platform-socket.io</code>
- [ ] <code>@nestjs/platform-ws</code>
- [ ] <code>@nestjs/testing</code>
- [ ] <code>@nestjs/websockets</code>
- [ ] Other (see below)

### NestJS version

10.3.0

### Packages versions

```
[System Information]
OS Version     : macOS Unknown
NodeJS Version : v20.10.0
PNPM Version    : 8.11.0 

[Nest CLI]
Nest CLI Version : 10.2.1 

[Nest Platform Information]
platform-express version : 10.3.0
schematics version       : 10.0.3
testing version          : 10.3.0
common version           : 10.3.0
core version             : 10.3.0
cli version              : 10.2.1

```

### Node.js version

20.10.0

### In which operating systems have you tested?

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by kamilmysliwiec** [comment]
https://docs.nestjs.com/recipes/swc#monorepo

**Comment by clintonb** [comment]
@kamilmysliwiec yes, I’ve seen those docs. My real ask is: document _why_ `swc` cannot be used without Webpack, and explicitly state that monorepo+swc does not work. It would be even better if the CLI threw an exception to avoid folks’ thinking they did something wrong. 

For added context: I am using Nest as part of a Temporal worker. Webpack’s bundling doesn’t play nicely with some of Temporal’s requirement resolution for data converters, and I honestly don’t have a real need for Webpack. 

I was trying to keep my project as close to the documented CLI as possible to aid future onboarding, but ultimately opted to call swc with some custom scripting.

**Comment by gabrielbull** [comment]
@clintonb I agree with you, it is not even necessary to have webpack to get a monorepo working with SWC, but you have to manually compile the other projects using custom scripts. I have raised another ticket on the nest-cli project but it was closed and locked: https://github.com/nestjs/nest-cli/issues/2645

You can see a working example of a NestJS monorepo using SWC without webpack here: https://github.com/gabrielbull/nestjs-monorepo-swc

This can be started using this command:
```
yarn workspace project start:workaround
```

Basically, only running SWC to compile the lib package manually. Here's what I'm basically doing:

```
swc --config-file .swcrc --out-dir ./dist --strip-leading-paths --extensions .js,.ts ../lib/src
nest start
```

**Comment by kamilmysliwiec** [comment]
SWC doesn't have a built-in modules resolution system (like Webpack or Rollup). See https://github.com/swc-project/swc/discussions/5590#discussioncomment-6524432 If we wanted to support this in NestJS CLI, we would need to implement it ourselves, that's why Webpack is the recommended way to go. If someone really can't use Webpack in their project, custom scripting is the recommended approach. However, remember that by default, Webpack is configured to _not bundle_ the `node_modules` folder (when used through NestJS CLI), so it shouldn't be a problem for most projects (it only wires up application code and local libraries from the `libs` folder).
